src/client/mod.rs

```diff
@@ -359,7 +346,7 @@ impl<'a> Future for ClientBuilder<'a> {
             let event_handler = self.event_handler.take();
             let raw_event_handler = self.raw_event_handler.take();
             let intents = self.intents;
-            let http = Arc::new(self.http.take().unwrap());
+            let http = Arc::new(std::mem::replace(&mut self.http, Default::default()));
```

Comment:
[`std::mem::take`][take] can be used instead. It performs same work as this `replace` call here, but is simpler to use and shortens the code a bit. 
```suggestion
            let http = Arc::new(std::mem::take(&mut self.http));
```

[take]: https://doc.rust-lang.org/stable/std/mem/fn.take.html